src/components/Dialogs/Conditions/ConditionsManager.js

```diff
@@ -0,0 +1,165 @@
+import i18n from '@dhis2/d2-i18n'
+import { Button, IconInfo16 } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+import { connect } from 'react-redux'
+import { tSetCurrentFromUi } from '../../../actions/current'
+import { tSetUiConditionsByDimension } from '../../../actions/ui'
+import {
+    parseConditionsArrayToString,
+    parseConditionsStringToArray,
+} from '../../../modules/conditions'
+import { sGetMetadata } from '../../../reducers/metadata'
+import {
+    sGetDimensionIdsFromLayout,
+    sGetUiConditionsByDimension,
+} from '../../../reducers/ui'
+import DimensionModal from '../DimensionModal'
+import NumericCondition from './NumericCondition'
+import classes from './styles/ConditionsManager.module.css'
+
+const ConditionsManager = ({
+    conditions,
+    isInLayout,
+    onUpdate,
+    dimension,
+    onClose,
+    setConditionsByDimension,
+}) => {
+    const [conditionsList, setConditionsList] = useState(
+        parseConditionsStringToArray(conditions)
+    )
+
+    const addCondition = () => setConditionsList([...conditionsList, ''])
+
+    const removeCondition = id =>
+        setConditionsList(conditionsList.filter((_, index) => index !== id))
+
+    const setCondition = (id, value) =>
+        setConditionsList(
+            conditionsList.map((condition, index) =>
+                index === id ? value : condition
+            )
+        )
```

Comment:
Perhaps consider renaming `id` to `conditionIndex` or sth, since all of this seems index based.